.drone.yml

```diff
@@ -59,13 +59,60 @@ steps:
         from_secret: drone_git_token
     commands:
       - git clone https://$${DRONE_GIT_USERNAME}:$${DRONE_GIT_TOKEN}@github.com/UKHomeOfficeForms/hof-services-config.git
+      - find hof-services-config/* | grep -v $HOF_CONFIG | grep -v 'infrastructure' | xargs rm -rf
     when:
       branch:
         include:
         - master
         - feature/*
       event: [push, pull_request]
 
+  - name: sonar_scanner_branch
+    <<: *sonar_scanner
+    when:
+      branch:
+        include:
+          - master
+          - feature/*
+      event: pull_request
+
+  - name: setup_branch
+    pull: if-not-exists
+    image: node:20
+    environment:
+      NOTIFY_KEY: USE_MOCK
+    commands:
+      - yarn install --frozen-lockfile
+      - yarn run postinstall
+    when:
+      branch:
+        include:
+          - master
+          - feature/*
+      event: pull_request
+
+  - name: linting_branch
+    <<: *linting
+    when:
+      branch:
+        include:
+          - master
+          - feature/*
+      event: pull_request
+    depends_on:
+      - setup_branch
+
+  - name: unit_tests_branch
+    <<: *unit_tests
+    when:
+      branch:
+        include:
+        - master
+        - feature/*
+      event: pull_request
+    depends_on:
+      - linting_branch
+
```

Comment:
rather than having two sets of steps that run the same task in different builds  e.g. unit
_tests _branch steps and unit_tests_deploy we could have one set that occur on both push and pull request events. So instead of sonar_scanner_branch, setup_branch, linting_branch, unit_tests_branch, setup_deploy, unit_tests_deploy, sonar_scanner_deploy, you could have just sonar_scanner, setup, linting and unit_tests and get rid of the duplicates